l2geth/rollup/sync_service.go

```diff
@@ -872,6 +876,17 @@ func (s *SyncService) applyTransactionToTip(tx *types.Transaction) error {
 	// The index was set above so it is safe to dereference
 	log.Debug("Applying transaction to tip", "index", *tx.GetMeta().Index, "hash", tx.Hash().Hex(), "origin", tx.QueueOrigin().String())
 
+	// Log transaction to the failover log
+	encodedTx := new(bytes.Buffer)
+	if err := tx.EncodeRLP(encodedTx); err != nil {
+		return err
+	}
+	if err := s.txLogger.Publish(s.ctx, encodedTx.Bytes()); err != nil {
+		pubTxDropCounter.Inc(1)
+		log.Error("Failed to publish transaction to log", "msg", err)
+		return fmt.Errorf("internal error: transaction logging failed")
```

Comment:
I don't love the idea of returning an error here because that means we tightly couple googles uptime with our ability to accept transactions. Not really sure of a good solution

Reply:
Yeah, I don't like quite like it either but it's the best option. With the way infra is setup right now, I don't think this will worsen failure rate of transaction submissions. i.e. the sequencer is [more likely](https://cloud.google.com/pubsub/sla) to fail than google pubsub. And going by actual historical [incidents](https://status.cloud.google.com/incident/cloud-pubsub), even less likely to fail.

Reply:
My biggest concern with this approach is the additional latency overhead used per tx for logging. The median time to publish a transaction is about 35ms. Not bad right now, but it's something to watch out if tx rate increases via gas fee reductions or by having more users.

Reply:
If we use this logic in two places as per my other comment, we should wrap this in a helper method in `rcfg/pub`.